core/pta/stats.c

```diff
@@ -137,6 +140,35 @@ static TEE_Result get_memleak_stats(uint32_t type,
 	return TEE_SUCCESS;
 }
 
+#if defined(CFG_TA_STATS)
+static TEE_Result get_user_ta_stats(uint32_t type,
+				    TEE_Param p[TEE_NUM_PARAMS])
+{
+	uint32_t res = TEE_SUCCESS;
+	/*
+	 * p[0].memref.buffer = output buffer to struct malloc_stats
+	 */
+	if (TEE_PARAM_TYPES(TEE_PARAM_TYPE_VALUE_INOUT,
+			    TEE_PARAM_TYPE_MEMREF_OUTPUT,
+			    TEE_PARAM_TYPE_NONE,
+			    TEE_PARAM_TYPE_NONE) != type) {
+		return TEE_ERROR_BAD_PARAMETERS;
+	}
+
+	if (p[0].value.a == STATS_INPUT_ZERO_SIZE)
+		res = tee_ta_instance_stats(NULL,
+					    &p[0].value.a);
+	else
+		res = tee_ta_instance_stats(p[1].memref.buffer,
+					    &p[1].memref.size);
+
+	if (res != TEE_SUCCESS)
+		EMSG("tee_ta_dump_stats return: 0x%x\n", res);
```

Comment:
note that `TEE_Result` are uint32_t by GPD TEE spec definition.
So in the `xMSG()` macros, use `PRIx32` to trace result code values. Here:
```c
		EMSG("tee_ta_dump_stats return: 0x%"PRIx32, res);
```

Note `xMSG()` always append a `\n` character to the message, so not needed (and removed in example above).

Reply:
good to know, thanks.